react-ui/src/applyFilter.js

```diff
@@ -10,19 +12,46 @@ export default async function applyFilter(imageData, filter) {
     canvas = canvas || document.createElement('canvas')
 
     const image = new Image()
-    image.onload = () => {
+    image.onload = async () => {
       const ctx = canvas.getContext('2d')
 
       canvas.width = image.width
       canvas.height = image.height
       ctx.drawImage(image, 0, 0, image.width, image.height)
 
       const imgData = ctx.getImageData(0, 0, canvas.width, canvas.height)
-      const withFilter = window.pixelsJS.filterImgData(imgData, filter)
 
-      ctx.putImageData(withFilter, 0, 0)
+      // Choose whether to use pixelsJS or styleTransfer
+      var is_style = (typeof filter != 'string')
```

Comment:
So this is sort of nasty - this function is doing very many things here:

1. Calculating image data from canvas (almost completely duplicated code from another file added here)
2. Reverse-engineering the option type to figure out which filter was passed in
3. Calling pixelJS or
4. Calling own function here

This needs to be broken into pieces.

Easiest way to do this is to:

1. Extract the dataurl parts.
2. Extract new code part from here into a well-named function.
3. Make each filter being passed in here have a `type` field - if `type` = 'pixelJS` then one transformer is called, otherwise yours is called.